core/src/main/scala/org/apache/pekko/persistence/cassandra/util/LazyFutureEval.scala

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.pekko.persistence.cassandra.util
+
+import org.apache.pekko.annotation.InternalApi
+
+import java.util.concurrent.atomic.AtomicReference
+import scala.concurrent.{ ExecutionContext, Future }
+import scala.util.Failure
+
+@InternalApi
+private[cassandra] object LazyFutureEval {
+  def apply[T](evalFunction: () => Future[T]): LazyFutureEval[T] = new LazyFutureEval[T](evalFunction)
+}
+
+/**
+ * An internal utility class that lazily creates an instance of T but that can recover if the evalFunction fails.
+ *
+ * It does not guarantee that the evalFunction succeeds only once.
+ *
+ * This class is not recommended for non-Pekko usage.
+ */
+@InternalApi
+private[cassandra] class LazyFutureEval[T](evalFunction: () => Future[T]) {
```

Comment:
This utility function looks like something that should be placed into Pekko core?

Reply:
I'm happy to move it if there is another Pekko use case but at the moment, I am unaware of another use case.

In general, the use of `lazy val` that has a `Future[T]` type should be replaced by something like this.

There is also an issue with how recovery should work. My current implementation is fairly simplistic and it might be better to use the Pekko scheduler to perform continuous retries (until we get a successful result).

Reply:
I find the name somewhat misleading, as the `lazy` keyword implies only one execution. This is more of a "at least once" character.

Reply:
renamed as RetryableFutureEval